### ISSUE TITLE
writePositionSource always throws IllegalStateException

### ISSUE THREAD

**Issue by cxzlw** [GeyserMC/MCProtocolLib#738]
```
java.lang.IllegalStateException: Unknown position source type!
        at com.github.steveice10.mc.protocol.codec.MinecraftCodecHelper.writePositionSource(MinecraftCodecHelper.java:526)
        at com.github.steveice10.mc.protocol.codec.MinecraftCodecHelper.writeParticleData(MinecraftCodecHelper.java:499)
        at com.github.steveice10.mc.protocol.packet.ingame.clientbound.level.ClientboundLevelParticlesPacket.serialize(ClientboundLevelParticlesPacket.java:57)
        at Replay.add(Replay.kt:71)
        at App$run$1.packetReceived(Main.kt:100)
        at com.github.steveice10.packetlib.tcp.TcpSession.callPacketReceived(TcpSession.java:157)
        at com.github.steveice10.packetlib.tcp.TcpSession.lambda$channelRead0$2(TcpSession.java:386)
        at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

Got this when a Sculk Sensor was activated.

**Comment by cxzlw** [comment]
This function always throws IllegalStateException, although positionSource is instance of BlockPositionSource. 
```java
    public void writePositionSource(ByteBuf buf, PositionSource positionSource) {
        this.writeResourceLocation(buf, positionSource.getType().getResourceLocation());
        if (positionSource instanceof BlockPositionSource) {
            this.writePosition(buf, ((BlockPositionSource) positionSource).getPosition());
        } else if (positionSource instanceof EntityPositionSource) {
            this.writeVarInt(buf, ((EntityPositionSource) positionSource).getEntityId());
            buf.writeFloat(((EntityPositionSource) positionSource).getYOffset());
        }

        throw new IllegalStateException("Unknown position source type!");
    }
```
from [MinecraftCodecHelper.java:517](https://github.com/GeyserMC/MCProtocolLib/blob/master/src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java#L416)

**Comment by cxzlw** [comment]
pr #739

**Comment by cxzlw** [comment]
solved in ee030499b53500f1b2146e42c7a9639005489700
